lib/galaxy/webapps/galaxy/controllers/root.py

```diff
@@ -89,7 +89,7 @@ def login(self, trans, redirect=None, is_logout_redirect=False, **kwd):
                 len(trans.app.config.oidc) == 1 and
                 len(trans.app.auth_manager.authenticators) == 0
                 and is_logout_redirect is False):
-            provider = trans.app.config.oidc.keys()[0]
+            provider = list(trans.app.config.oidc.keys())[0]
```

Comment:
Not a performance issue here, but you can skip the (potentially expensive) list creation with:
```suggestion
            provider = next(iter(trans.app.config.oidc.keys()))
```